lib/core/proxy.c

```diff
@@ -373,10 +373,10 @@ static int on_body(h2o_http1client_t *client, const char *errstr)
 
 static char compress_hint_to_enum(const char *val, size_t len)
 {
-    if (!strncasecmp("ON", val, len)) {
+    if (h2o_lcstris(H2O_STRLIT("on"), val, len)) {
```

Comment:
The order of the argument should be `h2o_lcstris(val, len, H2O_STRLIT("on"))`. The function only converts the first set of arguments to lowercase.